app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_results = {}
+
+    def wrapper(*args: (int, float, bool, str, bytes, tuple)) -> Any:
```

Comment:
You could add a decorator from `functools` to preserve the wrapped function name and docstring.